feat: add de translation

starfetch opened the pull request:
Hi! 
I've started on a German translation for the strings, currently it's still a WIP but should be done soon enough.
Opened this draft PR to avoid possibly doubling the translation efforts.

So far I've created a `values-de` folder and added the English `strings.xml`, now I'm slowly replacing the English strings with the German ones.
I hope that this is the correct way to go, if not, happy to change and adjust.

starfetch committed:
fest: initial copy of en

starfetch committed:
feat: First quick translations

starfetch committed:
fest: more translations

starfetch committed:
feat: more translations

starfetch committed:
feat: final translations

starfetch committed:
fix: typos

starfetch commented:
and I'm done! Ready for review.

Moustachauve commented:
Thank you! This looks great. I asked in the WLED discord german channel if someone could help me review this :)

konradmoesch commented:
As a native german speaker, I have tested your branch. The texts look good, but some minor additional changes were needed to integrate and ship the new language with the app.
With those additions, this PR looks LGTM to me

konradmoesch commented:
Since those changes are needed outside of `values-de/strings.xml`, I will post them here:
- `app/src/main/res/xml/locales_config.xml`:
Add `<locale android:name="de"/>`
- `app/build.gradle`(l.22):
`resourceConfigurations += ['de', 'en', 'fr']`

konradmoesch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+    <string name="no_compatible_version_found_details">Es würde keine Version gefunden, die mit deinem Gerät kompatibel ist. Du musst dein Gerät manuell updaten.</string>
+    <string name="device_offline">Gerät Offline</string>
+    <string name="show_error">Zeige Fehler</string>
+    <string name="hide_error">Verberge Fehler</string>
```
"Fehler anzeigen" and "Fehler verbergen" maybe? Both options should be fine, though

starfetch replied:
I also prefer your suggestions.

konradmoesch reviewed (approved):
I found one typo.
Everything else are just alternate phrasing options, where both (yours and my suggestions) should be fine, I think.
Thanks for the patch, I really like using the app in german:)

starfetch commented:
Hi @Moustachauve, I'll gladly take a look, and thanks @konradmoesch for the review (apologies for completely missing your feedback).

starfetch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+    <string name="awkward_akemi_character">Awkward Akemi Charakter</string>
+    <string name="device_list">Meine Geräte</string>
+    <string name="new_device_found">Neues Gerät erkannt</string>
+    <string name="device_view">Geräte Ansicht</string>
```
Agreed, good catch.

starfetch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+    <string name="description_brightness_slider">Helligkeitsregler</string>
+    <string name="description_menu_button">Öffne Menü</string>
+    <string name="description_back_button">Gehe zurück</string>
+    <string name="description_list_of_discovered_devices">Liste an erkannten Geräten</string>
```
Yep, that's smoother.

starfetch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+    <string name="privacy">Privatsphäre</string>
+    <string name="send_crash_data_to_developer">Sende Absturzdaten an Entwickler</string>
+    <string name="send_performance_data_to_developer">Sende Performancedaten an Entwickler</string>
+    <string name="device_ap_mode">WLED AP Modus</string>
```
A bit of hyphenation never hurt anyone (and makes it more clear what the `AP` belongs to).

starfetch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+    <string name="device_update_disclaimer_part2">Wenn jemand dieses Gerät für dich installiert hat, solltest du diese Person vielleicht fragen, ob es in Ordnung ist, dieses Gerät auf eine neue Version zu updaten.</string>
+    <string name="device_update_disclaimer_part3">WLED und WLED Native sind nicht dafür verantwortlich, wenn etwas wegen eines Updates schiefläuft.</string>
+    <string name="install_update">Update wird installiert</string>
+    <string name="ota_install_failed_device_locked">Dein Gerät könnte durch eine PIN geschützt sein oder mit einer Passphrase verschlüsselt.</string>
```
You're absolutely correct.

starfetch commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,97 @@
+        Von <xliff:g example="v0.13.0" id="old_version">%1$s</xliff:g> zu <xliff:g example="v0.14.0" id="new_version">%2$s</xliff:g>
+    </string>
+    <string name="see_update_details">Siehe Update</string>
+    <string name="checking_progress_update">Am überprüfen…</string>
```
Yes, not only better but also the only grammatically correct way.

starfetch committed:
feat: add feedback

starfetch committed:
feat: add DE translation

starfetch committed:
feat: add DE translation

starfetch commented:
@Moustachauve I have addressed the open feedback (thank you @konradmoesch for being so thorough). I'm not sure how best to proceed with the upstream changes, could you give me a few pointers?

Moustachauve committed:
Merge branch 'main' into pr/12

Moustachauve committed:
Delete build.gradle

It is no longer used, this was leftover from bad merge by me.

Moustachauve committed:
feat(i18n): Add German translations

This commit introduces German (de) localization to the app.

- Adds the `de` locale to `localeFilters` in `build.gradle.kts`.
- Adds a new `values-de/strings.xml` file with translated strings.
- Updates and adds new strings to the German translation file for better coverage.

Moustachauve commented:
/gemini review

Moustachauve committed:
Merge branch 'main' into pr/12

Moustachauve committed:
feat: Add German translations for device error

This commit adds German translations for the device error title, message, and the retry connection button.

Moustachauve committed:
Improve German translation for offline status

Changed "vor" (ago) to "seit" (since) in offline status strings to improve grammatical accuracy.

This is based on a suggestion by roll0r on Discord.

Moustachauve committed:
Merge branch 'main' into pr/12

Moustachauve committed:
feat: Add German translations for download handling

This commit adds new German string resources for various download-related messages, including states for completion, failure, and unsupported operations. It also improves an existing download message by adding a placeholder for the filename.

New strings cover:
- Download completion and failure statuses
- Unsupported download schemes and older Android versions
- File creation and saving errors
- Invalid data URI formats
- Actions like "Open" and "Dismiss"

Moustachauve committed:
Merge branch 'dev' into pr/12

# Conflicts:
#	app/src/main/res/xml/locales_config.xml

Moustachauve committed:
Merge branch 'dev' into pr/12

Moustachauve committed:
feat: Add German translations for widget and deep linking

This commit introduces German translations for several new features:
- Widget-related texts, including description and configuration prompts.
- Deep linking status and error messages.
- A notice for hidden devices.
